### PR TITLE
chore: remove deprecated winit method

### PR DESCRIPTION
**Pull request by wusyong**
Part of #34522 
winit's [EventLoop::run](https://docs.rs/winit/latest/winit/event_loop/struct.EventLoop.html#method.run) has been deprecated. It's only a matter of time before we update to the `run_app` method.

I tried to make sure it can still work in both headed and headless window. So WPT can still work. However, there are some errors on macOS and Linux when exiting in headed mode because the lifetime of `app`, `window`, and `servo` are a bit different than before. In [Verso](https://github.com/versotile-org/verso/blob/main/src/verso.rs#L52), I refactored a lot and even switched to glutin. I'm not sure we are going this path here. Could we open regression issues after this PR? We can discuss how to fix those issues. Otherwise, it will still need quite a lot of refactoring, even just for workarounds.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] There are tests for these changes. Existing WPT test should remain the same.

### CODE REVIEW

**Comment by wusyong** [comment]
@mukilan Simply closing the window should get the segfault on Linux. IIRC, on wayland, it's because double free the wayland surface after backtracking. I forgot what happened to x11, but I'll tried to give more details and log tomorrow.

File: ports/servoshell/desktop/cli.rs [hunk 1]
```diff
@@ -99,12 +100,18 @@ pub fn main() {
         None
     };
 
-    App::run(
+    let events_loop = EventsLoop::new(opts::get().headless, opts::get().output_file.is_some())
```
Comment (gterzian) [inline ports/servoshell/desktop/cli.rs:103]: nit `event_loop` since there is only one?

**Review by gterzian — APPROVED** [review]
LGTM, I tried running tests and a normal run on mac os and it seemed to work without the crash mentioned. 

I think not relying on deprecated method is a good thing, as long as we can keep using Servo normally for testing while the follow-up issues to be filed are fixed.

**Comment by Loirooriol** [comment]
The merge queue is working again.

**Comment by jdm** [comment]
Well, at least we know the smoketest is working as intended:
```
Run python mach smoketest --release
Servo exited with non-zero status 3221225477
```

**Comment by wusyong** [comment]
But previous windows smoke tests are fine. Some queue errors are reproducable in main branch sometimes. Not sure if they are intermittent. I'll rebase and check again.

**Comment by mrobinson** [comment]
I'm seeing this segfault also when running WPT tests: https://github.com/servo/servo/pull/34581#issuecomment-2554324784

Can we please revert this change until the segfault is fixed?

**Comment by wusyong** [comment]
@mrobinson The headless mode shouldn't encounter any segfault. It will only appear when you manually open a single test which will open in headed mode. Are you sure the crash is caused by closing the window in headed mode?

**Pull request merged**